library/ssl_msg.c

```diff
@@ -955,32 +961,38 @@ int mbedtls_ssl_encrypt_buf( mbedtls_ssl_context *ssl,
                             padlen + 1 ) );
 
 #if defined(MBEDTLS_USE_PSA_CRYPTO)
-        status = psa_cipher_encrypt_setup( &cipher_op,
-                                     transform->psa_key_enc, transform->psa_alg );
+        /* Skip psa encryption for null cipher */
+        if ( transform->psa_alg != MBEDTLS_SSL_NULL_CIPHER )
```

Comment:
TLS 1.2 divides ciphers in 3 categories:
- block (which in practice means CBC)
- stream (in practice, RC4 or NULL) 
- AEAD (in practice, GCM, CCM or ChachaPoly)

Since the special NULL "cipher" is in the stream category, we don't need to care about it here in the CBC branch (we tested for `mode == MBEDTLS_MODE_CBC` on line 900, I know this function is too long and hard to follow, I've been meaning to refactor it for years but never got the opportunity).